impeller/renderer/backend/metal/context_mtl.mm

```diff
@@ -376,4 +381,28 @@ new ContextMTL(device, command_queue,
   return buffer;
 }
 
+void ContextMTL::StoreTaskForGPU(std::function<void()> task) {
+  tasks_awaiting_gpu_.emplace_back(std::move(task));
+  while (tasks_awaiting_gpu_.size() > kMaxTasksAwaitingGPU) {
```

Comment:
Why do we need a maximum limit of tasks? If we do have a limit, do we need a mechanism to drop them gracefully?

For example, if there is a Future pending from toByteData, I'd think that dropping the task should cause an error to be thrown?

Reply:
`EncodeImageFailsWithoutGPUImpeller` tests this, it's handled gracefully since you have to check for the GPU in the callback and we never retry more than once.  Getting dropped is just like the behavior we had before we implemented the retry.

Reply:
Oh, so when we exceed the size we execute the task, and then the task sees the GPU is still unavailable and errors? makes sense. I would document that a bit.

Reply:
Added that info do the Context::StoreTaskForGPU docstring.